### PR TITLE
adding camera utils and test script

### PR DESCRIPTION
**Pull request by amandlek**
Adding new utility functions to work with camera and depth images, including the following:
- retrieving camera extrinsics / intrinsics
- retrieving physical (unnormalized) depth maps
- transforming camera pixel locations into the world frame, using a depth map
- projecting world frame points into camera pixels

I also implemented a test script to verify that all functions are working properly.

### CODE REVIEW

File: robosuite/demos/demo_segmentation.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+def segmentation_to_rgb(seg_im):
+    """
+    Helper function to visualize segmentations as RGB frames.
+    NOTE: assumes that geom IDs go up to 255 at most - if not,
+    multiple geoms might be assigned to the same color.
+    """
+    # ensure all values lie within [0, 255]
+    seg_im = np.mod(seg_im, 256)
+
+    # colors = random_colors(N=256, bright=True)
```
Comment (amandlek) [inline robosuite/demos/demo_segmentation.py:37]: we can probably remove this and the random_colors function right? It was there for debugging purposes

File: robosuite/demos/demo_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    for i in range(300):
+        action = 0.1 * np.random.uniform(low, high)
+        obs, reward, done, _ = env.step(action)
+
+        if i % 5 == 0:
+            video_img = obs[f"{camera}_segmentation_{segmentation_level}"].squeeze(-1)[::-1]
+            np.savetxt("/tmp/seg_{}.txt".format(i), video_img, fmt="%.2f")
+            video_img = segmentation_to_rgb(video_img)
+            video_writer.append_data(video_img)
+            import json
```
Comment (amandlek) [inline robosuite/demos/demo_segmentation.py:97]: all of these lines were for debugging I think - we should remove them, and perhaps put an argument into this script for video path?

File: robosuite/utils/video_utils.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
+"""
+A set of utilities for generating nice robosuite videos :D
+
+NOTE: This requires cv2 package to be installed
```
Comment (amandlek) [inline robosuite/utils/video_utils.py:4]: are you sure this requires cv2?

File: robosuite/utils/video_utils.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+            for k, frame in self.grab_next_frame().items():
+                frames_dict[k].append(frame)
+
+            # Update camera pose
+            self.rotate_camera(point=pan_point, axis=pan_axis, angle=pan_rate)
+
+        # Stack all frames and return
+        return {k: np.stack(frames) for k, frames in frames_dict.items()}
+
+    def modify_xml_for_camera_movement(self, xml, camera_name):
```
Comment (amandlek) [inline robosuite/utils/video_utils.py:239]: these seem to be copied from the camera movement script. We should probably avoid duplication, and change that script to use this function instead

File: robosuite/demos/demo_segmentation.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    options["robots"] = ["Panda"]
+
+    # Choose controller
+    controller_name = "OSC_POSE"
+
+    # Choose camera
+    camera = "agentview"
+
+    # Choose segmentation type
+    segmentation_level = "instance"         # Options are {instance, class, element}
```
Comment (amandlek) [inline robosuite/demos/demo_segmentation.py:64]: can we expose this as an argument, to easily demonstrate the different modes in this script?

**Review by amandlek — COMMENTED** [review]
LGTM in general, just proposed some minor changes

**Comment by yukezhu** [comment]
Please add a short description at https://github.com/ARISE-Initiative/robosuite/blob/master/docs/demos.md for `demo_segmentation.py`. Thanks!

File: robosuite/utils/video_utils.py (hunk 4 above)
Comment (cremebrule) [reply robosuite/utils/video_utils.py:239]: Agreed. What exact script are you referring to? I don't recall adding a script with duplicate functionality

File: robosuite/utils/video_utils.py (hunk 3 above)
Comment (cremebrule) [reply robosuite/utils/video_utils.py:4]: great catch! removed

**Review by yukezhu — CHANGES_REQUESTED** [review]
This PR broke the tests. `tests/test_environments/test_all_environments.py` failed in this branch, but not in master.

**Review by yukezhu — APPROVED** [review]
LGTM!

**Pull request merged**